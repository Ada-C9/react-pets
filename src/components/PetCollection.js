import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

import NewPetForm from './NewPetForm';
import PetTable from './PetTable';

const PETS_URL = 'https://petdibs.herokuapp.com/pets';

class PetCollection extends Component {
  static propTypes = {
    updateStatusCallback: PropTypes.func.isRequired
  };

  constructor() {
    super();

    // Set state to the static data from props
    this.state = {
      pets: []
    };
  }

  componentDidMount() {
    this.props.updateStatusCallback('Loading pets...', 'success');

    axios.get(PETS_URL)
      .then((response) => {
        console.log('Success!');
        console.log(response);

        this.props.updateStatusCallback('Successfully loaded pets!', 'success');

        // Do some pre-processing on the data
        const pets = response.data.slice(0, 100);
        this.setState({ pets: pets });
      })
      .catch((error) => {
        // console.log('Error :(');
        // console.log(error);

        // Get something on the screen so the user knows
        this.props.updateStatusCallback(error.message, 'error');

        // This is what we would have done w/ jQuery.
        // But, in React everything on the screen comes
        // through rendering.
        // $('.status').html(error.message)
      });
  }

  addPet = (pet) => {
    axios.post(PETS_URL, pet)
      .then((response) => {
        console.log(response);
        this.props.updateStatusCallback(`Successfully added pet ${ pet.name }!`, 'success');

        let updatedPets = this.state.pets;
        updatedPets.push(pet);

        this.setState({ pets: updatedPets });
      })
      .catch((error) => {
        this.props.updateStatusCallback(`Error adding pet ${ pet.name }.`, 'error');
      });
  }

  render() {


    return (
      <section>
        <PetTable pets={this.state.pets} />

        <NewPetForm addPetCallback={this.addPet}/>
      </section>
    );
  }
}

export default PetCollection;
