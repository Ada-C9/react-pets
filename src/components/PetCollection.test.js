import React from 'react';
import PetCollection from './PetCollection';
import { mount, shallow } from 'enzyme';

describe('PetCollection', () => {
  test('deep mount', () => {
    const petCollection = mount(
      <PetCollection updateStatusCallback={()=>{}} />
    );

    expect(petCollection).toMatchSnapshot();

    petCollection.unmount();
  });

  test('shallow mount', () => {
    const petCollection = shallow(
      <PetCollection updateStatusCallback={()=>{}} />
    );

    expect(petCollection).toMatchSnapshot();

    
  });
});
