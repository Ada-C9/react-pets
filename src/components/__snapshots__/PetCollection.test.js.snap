// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetCollection deep mount 1`] = `
<PetCollection
  updateStatusCallback={[Function]}
>
  <section>
    <table>
      <tbody>
        <tr>
          <th>
            Name
          </th>
          <th>
            Breed
          </th>
          <th>
            Age
          </th>
          <th>
            About
          </th>
        </tr>
      </tbody>
    </table>
    <NewPetForm
      addPetCallback={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <div>
          <label
            htmlFor="name"
          >
            Name
          </label>
          <input
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            htmlFor="breed"
          >
            Breed
          </label>
          <input
            name="breed"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            htmlFor="age"
          >
            Age
          </label>
          <input
            name="age"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            htmlFor="about"
          >
            About
          </label>
          <input
            name="about"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <input
            type="submit"
          />
        </div>
      </form>
    </NewPetForm>
  </section>
</PetCollection>
`;

exports[`PetCollection shallow mount 1`] = `
<section>
  <table>
    <tbody>
      <tr>
        <th>
          Name
        </th>
        <th>
          Breed
        </th>
        <th>
          Age
        </th>
        <th>
          About
        </th>
      </tr>
    </tbody>
  </table>
  <NewPetForm
    addPetCallback={[Function]}
  />
</section>
`;
